src/app-logic/url-handling.js

```diff
@@ -489,9 +489,18 @@ type Location = {
   hash: string,
 };
 
+/**
+ * Create url state from window.location.
+ *
+ * `profile` parameter is nullable and optional. It's nullable because data sources
+ * like from-addon can't upgrade a url for a freshly captured profile. So we need
+ * to skip upgrading for these sources. It's also optional for only testing purpose.
+ * That way we won't have to create a profile for every time we want to create a
+ * state from location. Do not use it except testing.
+ */
 export function stateFromLocation(
   location: Location,
-  profile?: Profile
+  profile?: Profile | null
```

Comment:
Suggestion: "It's also optional for only testing purpose" appears to be an incorrect statement.`src/components/app/UrlManager.js` is using it without the second parameter.

Reply:
Ah, thanks for pointing out. I missed that for some reason. I changed the comment to reflect that.